src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -197,14 +204,34 @@ private boolean cleanSubscriptionRepeatFlagHandlerDb(String subscriptionNameQuer
      *            name of the current user
      * @return a String object
      */
-
     private String generateQuery(String subscriptionName, String ldapUserName) {
         String query = String.format(SUBSCRIPTION_NAME, subscriptionName);
-        if (!ldapUserName.isEmpty()) {
+        if (!ldapUserName.isEmpty() && ldapUserName != null) {
```

Comment:
Shouldn't this if statement be reversed?, since "ldaUseName" can be null when isEmpty() is executed.
Should be reversed to:
"if (ldapUserName != null && !ldapUserName.isEmpty())"

Reply:
You are right... Null check should be first

Fixed